tslearn/preprocessing.py

```diff
@@ -83,7 +83,13 @@ def __init__(self, min=0., max=1.):
         self.min_ = min
         self.max_ = max
 
-    def fit_transform(self, X, **kwargs):
+    def fit(self, X, y=None, **kwargs):
```

Comment:
Do we really need `fit` and `transform` methods here? It seems `TransformerMixin` does not require it: https://scikit-learn.org/stable/modules/generated/sklearn.base.TransformerMixin.html

Reply:
It seems to be required for some scenarios. The transform would be required for example in GridSearchCV on a Pipeline for evaluation (https://scikit-learn.org/stable/developers/contributing.html#pipeline-compatibility)

Reply:
Ok, then the doc for it should be explicit that it is exactly the same as `fit_predict` no?

And what about fit? Is it required? Because in this case, a `fit` method is supposed to do.... just nothing I think (just return `self`?).

Reply:
I will check why I exactly included this, and leave it out in case it would be unnecessary! I will indeed create some docs that say that there methods are all the same!

Reply:
`fit` should return `self`, `transform` should do the transformation and `fit_transform` is not needed (inheriting from `TransformerMixin` provides the `fit_transform` as `fit().transform()`:
https://github.com/scikit-learn/scikit-learn/blob/7813f7efb5b2012412888b69e73d76f2df2b50b6/sklearn/base.py#L526-L556

You can override it if there's a more efficient implementation of `fit_transform` than `fit().transform()`, but in this case, since `fit` does nothing and returns `self`, `fit_transform` is redundant.

Reply:
I agree with @johannfaouzi on that point.